File: articles/decision-trees-and-structured-choice/r/run_all_decision_tree_workflows.R
# run_all_decision_tree_workflows.R
# Run all R workflows for Decision Trees and Structured Choice.

args <- commandArgs(trailingOnly = FALSE)
file_arg <- grep("^--file=", args, value = TRUE)

if (length(file_arg) > 0) {
  script_path <- normalizePath(sub("^--file=", "", file_arg[1]), mustWork = TRUE)
  article_root <- normalizePath(file.path(dirname(script_path), ".."), mustWork = TRUE)
} else {
  article_root <- getwd()
}

r_dir <- file.path(article_root, "r")

scripts <- c(
  "decision_trees_structured_choice_workflow.R",
  "rollback_profiles.R",
  "value_of_information_report.R",
  "probability_sensitivity_profiles.R",
  "threshold_analysis_tables.R",
  "regret_summary_tables.R"
)

for (script in scripts) {
  path <- file.path(r_dir, script)
  message("Running ", path)
  source(path)
}

message("All R decision-tree workflows completed.")

File: articles/decision-trees-and-structured-choice/python/run_all_decision_tree_workflows.py
#!/usr/bin/env python3
"""Run all Python workflows for Decision Trees and Structured Choice."""

from __future__ import annotations

from pathlib import Path
import subprocess
import sys

ARTICLE_ROOT = Path(__file__).resolve().parents[1]
PYTHON_DIR = ARTICLE_ROOT / "python"

SCRIPTS = [
    "expected_value_rollback.py",
    "chance_node_evaluator.py",
    "decision_node_selector.py",
    "value_of_information_analysis.py",
    "probability_threshold_analysis.py",
    "regret_profile_analysis.py",
    "review_trigger_generator.py",
    "decision_record_exporter.py",
    "decision_trees_structured_choice_simulation.py",
]


def main() -> None:
    for script in SCRIPTS:
        path = PYTHON_DIR / script
        print(f"Running {path.name}...")
        subprocess.run([sys.executable, str(path)], check=True)
    print("All Python decision-tree workflows completed.")


if __name__ == "__main__":
    main()
